tools/NamedPipeCmd/main.go

```diff
@@ -62,32 +80,21 @@ func main() {
 		return
 	}
 
-	fprintf("connected")
+	fprintf("connected.\n")
 
 	defer func() {
 		fprintf("\nClosing named pipe...\n")
 		pipeClient.Close()
 
 	}()
 
-	// The HTTP client still needs the URL as part of the request even though
-	// the underlying transport is a named pipe.
-	method := os.Args[1]
-	uriPath := os.Args[2]
-	url := "http://localhost" + uriPath
-
-	if (*method != "GET") && (*method != "POST") {
-		fprintf("Invalid HTTP method: %s\n", *method)
-		os.Exit(1)
-	}
-
 	// This HTTP client handles formatting and chunked messages.
-	fmt.Printf("Creating HTTP client. ")
+	fprintf("Creating HTTP client. ")
 	httpClient := http.Client{
 		Timeout: 10 * time.Second,
 		Transport: &http.Transport{
 			MaxIdleConns:    2,
-			IdleConnTimeout: 30 * time.Second,
+			IdleConnTimeout: 5 * time.Second,
```

Comment:
would there be any reason to make this a command line arg too? (with 5 as the default)

Reply:
Not really. It makes sense if the transmission was over a network, but for not for local IPC.  The idle time starts the moment the named pipe transmission is done; it is holding the named pipe open for no good reason.  If there are resource conflicts, then the system as a whole would be suffering other symptoms too.